### PR TITLE
Fix issue with minigraph node navigation

### PR DESCRIPTION
**Pull request by jshaughn**
Protect against minigraph problems when the parent detail page component is changing its target element.

This is a semi-hack to prevent a major refactoring of how we work with the GraphDatasource object in the detail pages. Because we re-use the GraphDatasource, and it's an asynch fetch, it's possible that if the component gets re-used for a new workload/service/app, that things get out of sync.

In short, detect a mismatch between the target component and the datasource, and avoid updating the graph until things get back in sync.

Also:
- remove an unwanted dependency on "graphData" in effect that responds to a change in selectedIds
- slightly shorten the focus node flashing.
- suppress a bunch of console warnings.

Fixes https://github.com/kiali/kiali/issues/7839

### CODE REVIEW

File: frontend/src/index.tsx [hunk 1]
```diff
@@ -54,4 +57,9 @@ if (!window.location.pathname.includes(rootBasename)) {
   router.navigate(`/${window.location.search}`, { replace: true });
 }
 
+// Suppress thousands of console warnings when scrolling over PFT graphs
+configure({
+  enforceActions: 'never'
+});
```
Comment (nrfox) [inline frontend/src/index.tsx:63]: Should we be suppressing these or are the warnings indicative of a problem with our implementation?

Comment (jshaughn) [reply frontend/src/index.tsx:63]: I think the issue is rooted in PFT.  We get thousands of warnings as we move the cursor over the graph.  It seems to be a PFT thing, not something we are doing.

Comment (nrfox) [reply frontend/src/index.tsx:63]: I can open an issue with PFT. I've noticed these warnings too and wondered about them but didn't do anything about it.

Comment (jshaughn) [reply frontend/src/index.tsx:63]: Before opening an issue, I guess I should ensure it's not somehow our problem.  I think we can leave it suppressed while we look into it.  I'm slight;y suspicious now, because when mousing around in their "surge" demo, I don't see the warnings.

Comment (nrfox) [reply frontend/src/index.tsx:63]: I only see the warnings in dev mode I don't see them with the production build of the UI.

Comment (jshaughn) [reply frontend/src/index.tsx:63]: Well, now I'm not sure.  This happens when we call `elem.setData({some new set of data})`.  But I'm not sure why that's a problem.  the message says `Since strict-mode is enabled, changing (observed) observable values without using an action is not allowed`.  So, maybe it's an issue in PFT's impl of `setData()`, because under the covers the `data` field is tied to a mobX Observable.  Maybe I'll ask about this in PFT Slack...

Comment (jshaughn) [reply frontend/src/index.tsx]: OK, well that was a pita, but I believe we've gotten rid of the warnings and now perform element updates in the way PFT wants us to do it.  This may give us a small boost in responsiveness as the user mouses across the graph, not sure, but now we batch together a bunch of highlighting updates.

code is pushed.

**Review by ferhoyos — CHANGES_REQUESTED** [review]
Thanks @jshaughn for fixing the Cypress tests! Glad to see the PFT warning has been removed as well. While testing the PR again, I found another place where the `setObserved` function needs to be called.

**GraphPF.tsx - line 545** and **Mesh.tsx - line 402**
```
if (e.getType() !== 'graph') {
        setObserved(() => {
           controller.removeElement(e);
        });
}
```

**Comment by jshaughn** [comment]
@ferhoyos good catch!  pushed...

**Review by ferhoyos — APPROVED** [review]
LGTM
